Documentation/bpf.rst

```diff
@@ -1822,6 +1995,313 @@ pointers. After that all the programs themselves are created through the BPF
 system call, and tail called maps, if present, updated with the program's file
 descriptors.
 
+bpftool
+-------
+
+bpftool is the main introspection and debugging tool around BPF and developed
+and shipped along with the Linux kernel tree under ``tools/bpf/bpftool/``.
+
+The tool can dump all BPF programs and maps that are currently loaded in
+the system, or list and correlate all BPF maps used by a specific program.
+Furthermore, it allows to dump the entire map's key / value pairs, or
+lookup, update, delete individual ones as well as retrieve a key's neighbor
+key in the map. Such operations can be performed based on BPF program or
+map IDs or by specifying the location of a BPF file system pinned program
+or map. The tool additionally also offers to pin maps or programs into the
+BPF file system.
+
+For a quick overview of all BPF programs currently loaded on the host
+invoke the following command:
+
+  ::
+
+     # bpftool prog
+     398: sched_cls  tag 56207908be8ad877
+        loaded_at Apr 09/16:24  uid 0
+        xlated 8800B  jited 6184B  memlock 12288B  map_ids 18,5,17,14
+     399: sched_cls  tag abc95fb4835a6ec9
+        loaded_at Apr 09/16:24  uid 0
+        xlated 344B  jited 223B  memlock 4096B  map_ids 18
+     400: sched_cls  tag afd2e542b30ff3ec
+        loaded_at Apr 09/16:24  uid 0
+        xlated 1720B  jited 1001B  memlock 4096B  map_ids 17
+     401: sched_cls  tag 2dbbd74ee5d51cc8
+        loaded_at Apr 09/16:24  uid 0
+        xlated 3728B  jited 2099B  memlock 4096B  map_ids 17
+     [...]
+
+Similarly, to get an overview of all active maps:
+
+  ::
+
+    # bpftool map
+    5: hash  flags 0x0
+        key 20B  value 112B  max_entries 65535  memlock 13111296B
+    6: hash  flags 0x0
+        key 20B  value 20B  max_entries 65536  memlock 7344128B
+    7: hash  flags 0x0
+        key 10B  value 16B  max_entries 8192  memlock 790528B
+    8: hash  flags 0x0
+        key 22B  value 28B  max_entries 8192  memlock 987136B
+    9: hash  flags 0x0
+        key 20B  value 8B  max_entries 512000  memlock 49352704B
+    [...]
+
+Note that for each command, bpftool also supports json based output.
```

Comment:
What's the JSON argument? (I don't think we necessarily need an example usage, but the actual argument name might be useful)